onadata/apps/data_migration/surveytree.py

```diff
@@ -109,3 +60,59 @@ def insert_field_into_group_chain(self, field, group_chain):
             parent = group_field
 
         parent.append(field)
+
+    def sort(self, xformtree):
+        """Sort XML tree fields according to the order provided by XFormTree"""
+        pattern = xformtree.get_el_by_path(xformtree.DATA_STRUCT_PATH)[0]
+        self._sort(self.root, pattern)
+
+    @staticmethod
+    def get_order(tree, pattern):
+        """Get order according to which fields in XML should be sorted"""
+        LAST_FIELDS = ['imei', 'meta']
+        order = XFormTree.children_tags(pattern)
+        # Put other fields into order list
+        order += [e.tag for e in tree if e.tag not in order]
+        for field in LAST_FIELDS:  # Ensure LAST_FIELDS are in the end of XML
+            order.remove(field) if field in order else None
+        order += LAST_FIELDS
```

Comment:
`XFormTree.children_tags` is doing a `map` as opposed to `filter` with some condition - so it does not exclude any of the tags. In our cases cleaning tags is removing meta prefix from tag names. Example: `{http://www.w3.org/1999/xhtml}head` -> `head`. It only occurs in tags that are related to `xform`, so when we parse submissions it's not necessary, at least right now submissions doesn't contain tag prefixes.

That being said, I think that we should be consistent and in both situations we might want to used cleaned tags - both to avoid confusion and to be safe in case of prefixes appearing. Thanks for spotting that, I'll make patches so that we use clean tags everywhere